src/util.js

```diff
@@ -0,0 +1,169 @@
+}
+
+function ensureDir(dir) {
+  dir = resolvePath(dir);
+  if (!fs.existsSync(dir)) {
+    console.log(`Creating ${color.path(dir)}`);
+    fs.mkdirSync(dir, { recursive: true });
+  }
+}
+
+function updateDepotTools() {
+  const depot_dir = DEPOT_TOOLS_DIR;
+  console.log(`Updating ${color.path(depot_dir)}`);
+  if (os.platform() === 'win32') {
+    depotExecFileSync({}, 'cmd.exe', ['/c', path.resolve(depot_dir, 'update_depot_tools.bat')]);
+  } else {
+    depotExecFileSync({}, path.resolve(depot_dir, 'update_depot_tools'));
+  }
+}
+
+function ensureDepotTools() {
+  const depot_dir = DEPOT_TOOLS_DIR;
+
+  // if it doesn't exist, create it
+  if (!fs.existsSync(depot_dir)) {
+    console.log(`Cloning ${color.cmd('depot_tools')} into ${color.path(depot_dir)}`);
+    const url = 'https://chromium.googlesource.com/chromium/tools/depot_tools.git';
+    childProcess.execFileSync('git', ['clone', '-q', url, depot_dir], { stdio: 'inherit' });
+    updateDepotTools();
+  }
+
+  // if it's been awhile, update it
+  const now = new Date();
+  const msec_per_day = 86400000;
+  const days_before_pull = 14;
+  const days_untouched = (now.getTime() - fs.statSync(depot_dir).mtimeMs) / msec_per_day;
+  if (days_untouched >= days_before_pull) {
+    updateDepotTools();
+    fs.utimesSync(depot_dir, now, now);
+  }
+}
+
+function depotOpts(config, opts = {}) {
+  // some defaults
+  opts = {
+    encoding: 'utf8',
+    stdio: 'inherit',
+    ...opts,
+  };
+
+  opts.env = {
+    // set these defaults that can be overridden via process.env
+    PYTHONDONTWRITEBYTECODE: '1', // depot needs it
+    DEPOT_TOOLS_METRICS: '0', // disable depot metrics
+
+    ...process.env,
+    ...config.env,
+    ...opts.env,
+  };
+
+  // put depot tools at the front of the path
+  const key = pathKey();
+  opts.env[key] = [DEPOT_TOOLS_DIR, process.env[key]].join(path.delimiter);
+
+  return opts;
+}
+
+function depotExecSync(config, cmd, opts_in) {
+  const opts = depotOpts(config, opts_in);
+  console.log(color.childExec(cmd, null, opts));
+  childProcess.execSync(cmd, opts);
+}
+
+function depotExecFileSync(config, exec, args, opts_in) {
+  if (exec === 'python' && !path.isAbsolute(args[0])) {
+    args[0] = path.resolve(DEPOT_TOOLS_DIR, args[0]);
+  }
+  const opts = depotOpts(config, opts_in);
+  console.log(color.childExec(exec, args, opts));
+  childProcess.execFileSync(exec, args, opts);
+}
+
+function getSCCacheExec(root) {
+  return path.resolve(root, 'src', 'electron', 'external_binaries', 'sccache');
+}
+
+function ensureSCCache(config) {
+  const sccache = getSCCacheExec(config.root);
+  const opts = { env: config.env, stdio: 'ignore' };
+
+  if (os.platform() === 'win32') {
+    console.debug(`Building on Windows -- skipping ${color.path(sccache)}`);
+    return;
+  }
+
+  try {
+    childProcess.execFileSync(sccache, ['--stop-server'], opts);
+  } catch (e) {
+    // it's OK for this to fail -- maybe it wasn't running
+  }
+
+  for (;;) {
```

Comment:
nit: personally tend to prefer `while (true)` for readability on intentional infinite loops

Reply:
Is this a personal preference or idiomatic JS? If the latter I'm fine to make the change.

Generally I use `;;` because then we _know_ that the system isn't wasting cycles re-evaluating `true` each time we go through the loop

Reply:
ah fair! no, i just think it's slightly less readable but there's no meaningful need to change :D